### PR TITLE
feat: implement DP-105 percolation dismissal/snooze memory

### LINKED ISSUE #234: Persist dismissal and snooze memory by partition scope
> Source catalog: docs/roadmap/v0.8.0-dynamic-percolation-issues.md

# Persist dismissal and snooze memory by partition scope

1. Title
- Persist dismissal and snooze memory by partition scope

2. Parent epic
- #225

3. Problem
- dismissal memory exists for nudges but not for generalized surfaced items.

4. Why this matters
- repeated low-value resurfacing erodes trust.

5. Proposal
- extend scope-keyed memory store for per-item dismiss/snooze state with expiry.

6. Implementation notes
- likely touched: `src/extension.ts`, new `src/percolation/memory.ts`
- store by `partitionScope + contextHash + itemKind`
- include retention pruning alignment

7. Acceptance criteria
- dismissed item does not reappear before eligible window
- memory clears safely on scope switch and retention prune

8. Test plan
- unit tests for TTL and scope partitioning

9. Telemetry / metrics impact
- add dismiss/snooze counters for percolated items

10. Risks / edge cases
- stale memory across contextHash collisions

11. Dependencies
- DP-103

12. Suggested labels
- `area:policy`, `v0.8.x`

13. Suggested milestone
- `v0.8.x-dynamic-percolation-tuning`

14. Mockup reference
- `docs/ux/dynamic-percolation-mockups.md#stable-emphasis-tokens-no-layout-thrash`
## Linked Epic
- #225

### PR DESCRIPTION
**Pull request by jkordish**
## Summary
Implements DP-105 (#234) from epic #225 by adding partition-scoped percolation dismissal/snooze memory with TTL-based re-eligibility.

## What Changed
- Added `src/percolation/memory.ts`:
  - normalized memory record parsing
  - deterministic record keys (`contextHash + itemKind`)
  - TTL-based suppression decisions
  - scoped store trimming/upsert helpers
- Integrated memory into percolation ranking flow in `src/extension.ts`:
  - candidate filtering by active memory before ranking
  - partition-scoped workspace-state storage (`tacos.percolationMemory.*`)
- Persisted per-item memory writes from existing UX actions:
  - `dismissNudge` now records percolation item dismissal memory (30m window)
  - `snoozeAuto` now records percolation item snooze memory (until selected snooze expiration)
- Retention alignment:
  - workspace retention pruning now prunes stale percolation memory records
  - workspace-scoped clearing now includes percolation memory keys
- Metrics additions:
  - `percolationDismissActions`
  - `percolationSnoozeActions`
  - wired through `MetricRecord`, CSV export, and recorded-metric detection

## Verification
- `npm run format:check`
- `npm run lint`
- `npm run compile`
- `npm test -- percolationMemory.test.ts percolationTypes.test.ts percolationSuppression.test.ts percolationRanking.test.ts percolationExplainability.test.ts metrics.test.ts`

## Issue Links
- Parent epic: #225
- Closes #234

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -2621,6 +2635,87 @@ async function setNudgeFeedbackForCurrentContext(
...
+  if (!Number.isFinite(until) || until <= now) {
+    return false;
+  }
+
+  const mode = resolveCompanionRuntimeMode(getConfig());
+  const primary = rankPercolationForSummary(summary, mode, {
+    context,
+    workspaceRoot,
+    now,
+    ignoreMemory: true,
```
[the remark this answers is not in the document]
Comment (jkordish) [reply src/extension.ts]: Addressed in commit bd889f1. `rememberPercolationPrimaryForSummary(...)` now records memory for the currently visible ranked primary candidate (active memory filters applied) instead of re-ranking with `ignoreMemory: true`, so dismiss/snooze targets the item the user actually saw.
